modules/core/src/smithy4s/http/HttpStatusCode.scala

```diff
@@ -36,13 +39,19 @@ object HttpStatusCode extends CachedSchemaCompiler.Impl[HttpStatusCode] {
   type Aux[A] = internals.HttpCode[A]
 
   def fromSchema[A](schema: Schema[A], cache: Cache): HttpStatusCode[A] = {
-    val dynamicErrorCodeVisitor =
+    val dynamicErrorCode =
       schema.compile(new HttpResponseCodeSchemaVisitor())
-    val staticVisitor =
+    val staticErrorCode =
       schema.compile(new internals.ErrorCodeSchemaVisitor(cache))
-    new HttpStatusCode[A] {
-      def code(a: A, default: Int): Int =
-        dynamicErrorCodeVisitor(a).orElse(staticVisitor(a)).getOrElse(default)
+
+    dynamicErrorCode match {
+      case NoResponseCode =>
+        (a: A, default: Int) => staticErrorCode(a).getOrElse(default)
```

Comment:
as you can see, when no member has `@httpResponseCode`, we completely elude the call